application-task-manager-ui/src/main/resources/TaskManager/Code/TaskMacroInit.xml

```diff
@@ -124,32 +124,43 @@
       <cache>long</cache>
     </property>
     <property>
-      <code>require(['jquery', 'xwiki-meta'], function ($, xm) {
-  let init = function () {
-    console.log('Event called');
-    let isTaskMacroModal = !!$('.task-reference').length;
-    if (isTaskMacroModal) {
-      let modal = $('.modal');
-      let idInput = modal.find("input[name='id']");
-      if (idInput.val()) return;
-      let createDateInput = modal.find("input[name='createDate']");
-      let creatorInput = modal.find("input[name='creator']");
-      idInput.val(Math.random().toString(16).slice(2));
-      createDateInput.val(new Date().toLocaleDateString('en-UK'));
-      creatorInput.val(xm.userReference.toString());
-      console.log('done with init-ing');
-    }
-  };
-  init();
-  console.log('I am here!!!');
-  $(document).on('shown.bs.modal', '.modal', init);
-});</code>
+      <code>require.config({
+  paths: {
+    'moment-setup': new XWiki.Document(XWiki.Model.resolve('TaskManager.Code.MomentJSConfig', XWiki.EntityType.DOCUMENT)).getURL('jsx', 'minify=false'),
+    'moment-jdateformatparser': "$services.webjars.url('moment-jdateformatparser', 'moment-jdateformatparser.min')"
+  }
+});
+require(['moment-setup'], function () {
+  require(['jquery', 'xwiki-meta', 'moment', 'moment-jdateformatparser'], function ($, xm, moment) {
+    let init = function () {
+      let isTaskMacroModal = !!$('.task-reference').length;
+      if (isTaskMacroModal) {
+        let modal = $('.modal');
+        let idInput = modal.find("input[name='id']");
+        if (idInput.val()) return;
+        let createDateInput = modal.find("input[name='createDate']");
+        let creatorInput = modal.find("input[name='creator']");
+        var dateFormat = moment().toMomentFormatString($(idInput).data('dateformat'));
+        idInput.val(Math.random().toString(16).slice(2));
+        createDateInput.val(moment(new Date()).format(dateFormat));
+        let creatorValue = xm.userReference.toString();
+        if (creatorValue.startsWith(xm.wiki + ':')) {
+          creatorValue = creatorValue.substring(xm.wiki.length + 1);
+        }
```

Comment:
You shouldn't play with the reference syntax. Use instead the ``EntityReference`` JavaScript API. Something like:

    XWiki.currentDocument.documentReference.relativeTo(new XWiki.WikiReference(XWiki.currentWiki)).toString()